### PR TITLE
add some backward ops shape analysis

### PR DESCRIPTION
**Pull request by Yancey1989**
add softmax_backward, nll_loss_backward, native_layer_norm_backward in shape analysis pass

related #473

### CODE REVIEW

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 1]
```diff
@@ -1726,19 +1735,56 @@ class ShapePropagator : public PropertyPropBase {
...
+        node->outputs()[0]->setType(type);
+        auto normalized_shape =
+            node->get<c10::List<int64_t>>(attr::normalized_shape).value();
+        const size_t axis = type->dim().value() - normalized_shape.size();
+        std::vector<ShapeSymbol> stat_shape;
+        int64_t dims = axis + type->dim().value();
+
+        for (const auto idx : c10::irange(axis)) {
+          auto dim = type->symbolic_sizes()[idx];
+          if (dim.is_static())
```
Comment (fortianyou) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:1771]: nit: if (not dim.is_static) ?

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 2]
```diff
@@ -1726,19 +1735,56 @@ class ShapePropagator : public PropertyPropBase {
...
+      if (auto type = input_type(0)) {
+        node->outputs()[0]->setType(type);
+        auto normalized_shape =
+            node->get<c10::List<int64_t>>(attr::normalized_shape).value();
+        const size_t axis = type->dim().value() - normalized_shape.size();
+        std::vector<ShapeSymbol> stat_shape;
+        int64_t dims = axis + type->dim().value();
+
+        for (const auto idx : c10::irange(axis)) {
+          auto dim = type->symbolic_sizes()[idx];
```
Comment (fortianyou) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:1770]: Nit: rename to dimSize?

File: pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp [hunk 3]
```diff
@@ -1726,19 +1735,56 @@ class ShapePropagator : public PropertyPropBase {
...
+          if (dim.is_static())
+            stat_shape.emplace_back(ShapeSymbol::newSymbol());
+          else
+            stat_shape.emplace_back(ShapeSymbol::fromStaticSize(dim.value()));
+        }
+        for (const auto idx : c10::irange(axis, type->dim().value())) {
+          (void)idx; // Suppress unused variable warning
+          stat_shape.emplace_back(ShapeSymbol::fromStaticSize(1));
+        }
+        SymbolicShape shape(stat_shape);
```
Comment (fortianyou) [inline pytorch_blade/src/compiler/jit/torch/shape_analysis.cpp:1780]: Nit: rename to  symblicShape?

**Review by fortianyou — APPROVED** [review]
LGTM, with just tiny comments

**Review by fortianyou — APPROVED** [review]
LGTM
